hiclass/LocalClassifierPerParentNode.py

```diff
@@ -224,3 +224,44 @@ def _fit_digraph(self, local_mode: bool = False, use_joblib: bool = False):
         self.logger_.info("Fitting local classifiers")
         nodes = self._get_parents()
         self._fit_node_classifier(nodes, local_mode, use_joblib)
+
+    def get_predict_proba(self, X):
+        """
+        Prediction probabilities for each class for the given data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            The input samples. Internally, its dtype will be converted
+            to ``dtype=np.float32``. If a sparse matrix is provided, it will be
+            converted into a sparse ``csr_matrix``.
+        Returns
+        -------
+        prediction_probabilities : dict or list of dicts if X contains multiple samples.
+            Each dict contains the node as its key and prediction probabilities for
+            the predicted classes as value.
+        """
+        prediction_probabilities = []
+        for sample in X:
```

Comment:
I have completely removed this function with this [commit](https://github.com/scikit-learn-contrib/hiclass/pull/106/commits/81817c3ce04594bc289ebe50fe9a9ef01a1d881c) since it is not needed anymore.